fennel.lua

```diff
@@ -401,12 +424,14 @@ end
 
 -- Creates Lua-friendly symbol from a single symbol by mangling it.
 local function symMangle(symb)
-    if luaKeywords[symb] then
+    if luaKeywords[symb] or symb:match('^%d') then
         symb = '_' .. symb
     end
-    return symb:gsub('[^%w_]', function(c)
+    local function escaper(c)
```

Comment:
Is there a particular reason to create this function within `symMangle` rather than in the scope above it?

Reply:
I have moved it back to how it was before as it was clearer.